src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1335,7 +1372,25 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                 // When MaximumRetryCount is not specified, the totalRequests is 1.
                 if (totalRequests > 1 && ShouldRetry(response.StatusCode))
                 {
-                    int retryIntervalInSeconds = WebSession.RetryIntervalInSeconds;
+                    float retryIntervalInSeconds = WebSession.RetryIntervalInSeconds;
+                    int exponent = WebSession.MaximumRetryCount - totalRequests + 1;
+
+                    switch (RetryMode)
+                    {
+                        case WebRequestRetryMode.Fixed:
+                            retryIntervalInSeconds = WebSession.RetryIntervalInSeconds;
+                            break;
+                        case WebRequestRetryMode.Exponential:
+                            retryIntervalInSeconds = MathF.Min(MathF.ScaleB(retryIntervalInSeconds, exponent), _maximumRetryIntervalInSeconds);
+                            break;
+                        case WebRequestRetryMode.ExponentialJitter:
+                            Random random = new();
+                            retryIntervalInSeconds = MathF.Min(MathF.ScaleB(retryIntervalInSeconds, exponent) * random.NextSingle(), _maximumRetryIntervalInSeconds);
+                            break;
+                        default:
+                            retryIntervalInSeconds = WebSession.RetryIntervalInSeconds;
+                            break;
+                    }
```

Comment:
```suggestion
                    int exponent = WebSession.MaximumRetryCount - totalRequests + 1;

                    Random random = new();
                    float retryIntervalInSeconds = RetryMode switch
                    {
                        WebRequestRetryMode.Exponential => MathF.Min(MathF.ScaleB(retryIntervalInSeconds, exponent), _maximumRetryIntervalInSeconds),
                        WebRequestRetryMode.ExponentialJitter => MathF.Min(MathF.ScaleB(retryIntervalInSeconds, exponent) * random.NextSingle(), _maximumRetryIntervalInSeconds),
                        WebRequestRetryMode.Fixed or _ => WebSession.RetryIntervalInSeconds
                    };
```
I think that a switch expression could be better here.
https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/operators/switch-expression

Reply:
I have accepted your suggestion in 87535df